girder/molecules/server/molecule.py

```diff
@@ -332,6 +335,25 @@ def stream():
             .param('output_format', 'The format to convert to', paramType='path')
             .errorResponse('Output format not supported.', 400))
 
+    @access.public
+    @autoDescribeRoute(
+            Description('Get an SVG representation of a molecule.')
+            .param('id', 'The id of the molecule', paramType='path')
+            .errorResponse('Molecule not found.', 404)
+            .errorResponse('Molecule does not have SVG data.', 404))
+    def get_svg(self, id):
```

Comment:
I wonder if it wouldn't be better to have reference to a file id instead of saving the svg string on the molecule model directly.

Reason being, the browser should be able to cache the image instead of making a new rest request every time we want the image, and also with an actual file link we wouldn't have to do "unsafe" things such as setting the innerHTML of an element in order to display the image

Reply:
I would like to stick SVG in the record for now, caching may not be great, but we can set the mime type and include in pages I think.